<!--
This launch file starts the velodne driver for both the horizontal (Lite)
and the vertical lidar (high res).
  - This is for VLP 16 lidars
  - Make sure the correct calibration files are included
 -->

<launch>
    <!-- For horizontal lidar - VLP 16 Lite: -->
    <group ns="lidar_h">
      <!-- declare arguments with default values -->
        <arg name="calibration" default="$(find velodyne_pointcloud)/params/VLP-16.yaml"/>
        <arg name="device_ip" default="192.168.1.201" />
        <arg name="frame_id" default="hvlp_link" />
        <arg name="manager" default="$(arg frame_id)_nodelet_manager" />
        <arg name="max_range" default="130.0" />
        <arg name="min_range" default="0.4" />
        <arg name="pcap" default="" />
        <arg name="port" default="2368" />
        <arg name="read_fast" default="false" />
        <arg name="read_once" default="false" />
        <arg name="repeat_delay" default="0.0" />
        <arg name="rpm" default="600.0" />
        <arg name="cut_angle" default="3.14159" />
        <arg name="use_gps_timestamps" default="false" />

      <!-- start nodelet manager and driver nodelets -->
        <include file="$(find velodyne_driver)/launch/nodelet_manager.launch">
          <arg name="device_ip" value="$(arg device_ip)"/>
          <arg name="frame_id" value="$(arg frame_id)"/>
          <arg name="manager" value="$(arg manager)" />
          <arg name="model" value="VLP16"/>
          <arg name="pcap" value="$(arg pcap)"/>
          <arg name="port" value="$(arg port)"/>
          <arg name="read_fast" value="$(arg read_fast)"/>
          <arg name="read_once" value="$(arg read_once)"/>
          <arg name="repeat_delay" value="$(arg repeat_delay)"/>
          <arg name="rpm" value="$(arg rpm)"/>
          <arg name="cut_angle" default="3.14159" /> <!-- Start scan behind lidar -->
          <arg name="use_gps_timestamps" value="$(arg use_gps_timestamps)" />
        </include>

      <!-- start cloud nodelet -->
        <include file="$(find velodyne_pointcloud)/launch/cloud_nodelet.launch">
          <arg name="calibration" value="$(arg calibration)"/>
          <arg name="manager" value="$(arg manager)" />
          <arg name="max_range" value="$(arg max_range)"/>
          <arg name="min_range" value="$(arg min_range)"/>
        </include>
      </group>

</launch>